fix - issue 90 : interceptor 수정

Jennorresothie opened the pull request:
1. `PostService` 게시판 확인하는 메소드 생성
	- `PostRepository`에 `findByBoardIdAndIdIsDeletedIsFalse(long boardId, long postId)` 생성
	- `PostService`에 `validatePost` 생성

2. 게시판 유무 확인 시, `PostService` 바라보도록 변경
3. uri에서 pathvariable null  체크후 , 게시판 게시글 관계 확인

ps. 예외처리 과정 설명및 코드 추천 필요
close #90

Jennorresothie committed:
fix - issue 92 : Post에 BoardId 필드 추가

PostEntity에 BoardId 필드 추가
- 생성자와 `update()` 파라미터로 추가

PostService에서 BoardId 필드 받는 코드 추가
- `findAll()` 파라미터로 추가
- `update()`의 정보 수정하는 부분에 파라미터 추가

PostController에서 BoardId 필드 받는 코드 추가
- `@PathVariable("boardId") long boardId` 삭제
	- `findPost`
	- `deletePost`

- `@PathVariable("boardId") long boardId` 추가
	- `updateBoard`

관련 Dto들 수정
- UpsertPostRequest
- UpsertPost
- Post

post test 코드 수정

Jennorresothie committed:
fix - issue 90 : interceptor 수정

1. `PostService` 게시판 확인하는 메소드 생성
	- `PostRepository`에 `findByBoardIdAndIdIsDeletedIsFalse(long boardId, long postId)` 생성
	- `PostService`에 `validatePost` 생성

2. 게시판 유무 확인 시, `PostService` 바라보도록 변경
3. uri에서 pathvariable null  체크후 , 게시판 게시글 관계 확인

jhchoi57 commented on src/main/java/com/jootcamp/superboard/common/intercepter/PostCheckInterceptor.java:
```diff
@@ -1,39 +1,51 @@
+        Map<String, String> pathVariables =
+                (Map<String, String>) request.getAttribute(HandlerMapping.URI_TEMPLATE_VARIABLES_ATTRIBUTE);
+
+        if (pathVariables != null) {
```
MapUtils.isEmpty()

jhchoi57 commented on src/main/java/com/jootcamp/superboard/post/service/PostService.java:
```diff
@@ -57,4 +58,10 @@ public void update(UpsertPost upsertPost, long postId) {
+    // 게시글과 게시판 검증
+    public void validatePost(long boardId, long postId){
+        postRepository.findByBoardIdAndIdIsDeletedIsFalse(boardId, postId)
+                .orElseThrow(()->new BadRequestException("게시판과 게시글을 확인해주세요"));
```
Not found 로 수정

jhchoi57 commented on src/main/java/com/jootcamp/superboard/common/intercepter/PostCheckInterceptor.java:
```diff
@@ -1,39 +1,51 @@
+
+                // board 안에 post 있는지 확인
+                if (boardId != null && postId != null) {
+                    postService.validatePost(boardId, postId);
```
``` java
if(!boardService.existsBoard(boardId)) throw BoardNotFoundException();

if(!postService.existsPost(boardId, postId)) throw PostNotFoundException();
```

jhchoi57 commented on src/main/java/com/jootcamp/superboard/post/service/PostService.java:
```diff
@@ -57,4 +58,10 @@ public void update(UpsertPost upsertPost, long postId) {
+
+    // 게시글과 게시판 검증
+    public void validatePost(long boardId, long postId){
+        postRepository.findByBoardIdAndIdIsDeletedIsFalse(boardId, postId)
```
``` java
postRespository.existsByBoardIdAndIdIsDeletedIsFalse(boardId, postId)
```

Jennorresothie replied:
AndIdIsDeleted 에서 사이에 And 빠진것 같습니다

jhchoi57 commented on src/main/java/com/jootcamp/superboard/post/service/PostService.java:
```diff
@@ -57,4 +58,10 @@ public void update(UpsertPost upsertPost, long postId) {
     }
+
+    // 게시글과 게시판 검증
+    public void validatePost(long boardId, long postId){
```
existsPost

Jennorresothie committed:
피드백 반영

Jennorresothie committed:
피드백 반영

- issue 92 rebase

- null check
	- !pathVariables.isEmpty()

- Not found 로 수정
	- PostNotFoundException() 로 수정

- boardService 에 existsBoard(boardId) 메소드 추가

- postService 에 validatePost -> existsPost으로 메소드명 변경

- PostRepository에 findByBoardIdAndIdIsDeletedIsFalse -> existsByBoardIdAndIdIsDeletedIsFalse 로 변경

Jennorresothie committed:
AndIdIsDeleted 에서 사이에 And 추가

Jennorresothie committed:
Merge branch 'main' into feature/90
